docs/ref/models/fields.txt

```diff
@@ -115,9 +115,12 @@ human-readable name. For example::
         ("GR", "Graduate"),
     ]
 
+``choices`` can also be defined as a callable that expects no arguments and
+returns any of the formats described above.
```

Comment:
I think a small example would be useful somewhere in this documentation, in particular showing that the callable is passed in without being called, i.e. `CharField(choices=get_choices)` and not `CharField(choices=get_choices())`. It may also be worth highlighting why you'd want to use a callable, e.g. delaying I/O operations until first required to avoid slowing down start up.

_(As an aside, this makes me also wonder if it's also worth testing that a callable with, e.g. `@functools.cache`, will work correctly - and perhaps mentioning that this is supported as a way of lazily evaluating the choices once.)_

Reply:
I noticed that there are no changes to the documentation for `choices` in the forms documentation. Do we need to add anything there or is there already something mentioned?

Reply:
Thanks Nick for taking a look into this. You have a valid point, we should add a valid use case for using callable choices to the docs. I'll work on that. I have tested and cached callables work OK, so I'll have an example added to the docs.

Regarding no changes for the forms documentation: accepting callables in ChoiceField is a documented feature [since long time ago, it was added in Django 1.8](https://docs.djangoproject.com/en/4.2/ref/forms/fields/#django.forms.ChoiceField.choices) so as far as I understand this work has no impact in form's `ChoiceField`.